Add 0.66.0 CHANGELOG entry (#5676)
prometheus-operator/prometheus-operator: CHANGELOG.md

## 0.66.0 / 2023-06-14

> After research, we noticed how the Config-Reloader container is over dimensioned regarding it's CPU resources. We decided to change the default CPU requirements from 100m to 10m. For most environments this won't make a difference, but if you still need to increase Config-Reloader CPU resources, you can do so by adding the flags `-config-reloader-cpu-request` and `-config-reloader-cpu-limit` to the operator container. 

Reviewer: (suggestion)
```suggestion
After research, we noticed how the default CPU requests of the config-reloader container were too high compared to the effective usage. We decided to decrease the default CPU requests from `100m` to `10m`. For most environments, this won't make a difference but if you need to increase the values, you can use the `-config-reloader-cpu-request` and `-config-reloader-cpu-limit` CLI arguments of the Prometheus operator. 
```

> * [CHANGE] Config-Reloader default CPU requirements lowered from 100m to 10m #5539

Reviewer: ```suggestion
* [CHANGE] Decrease the default CPU requests for the config-reloader container from `100m` to `10m`. #5539
```

> * [FEATURE] Add `spec.alertmanagerConfiguration.global.PagerdutyURL` to the `Alertmanager` CRD #5469

Reviewer: (nit) missing final dot. Same remark for the other entries.

```suggestion
* [FEATURE] Add `spec.alertmanagerConfiguration.global.PagerdutyURL` to the `Alertmanager` CRD. #5469
```

* [FEATURE] Add `spec.valumeMounts` to the `ThanosRuler` CRD #5541
* [FEATURE] Add `spec.remoteWrite.sendNativeHistograms` to `Prometheus` and `PrometheusAgent` CRDs #5564
* [FEATURE] Add `spec.tracingConfig` to `Prometheus` and `PrometheusAgent` CRDs #5591 #5640

> * [FEATURE] Add `followRedirects` in order to allow to change the default value in `RemoteRead` #5612

Reviewer: ```suggestion
* [FEATURE] Add `followRedirects` field to the `RemoteRead` configuration. #5612
```

> * [FEATURE] Add option to configure `automountServiceAccountToken` for `Alertmanager` CRD #5474

Reviewer: ```suggestion
* [FEATURE] Add `spec.automountServiceAccountToken` to the `Alertmanager` CRD #5474
```

> * [FEATURE] Add `keep_firing_for` to rules in `PrometheusRule` CRD #5651

Reviewer: ```suggestion
* [FEATURE] Support `keep_firing_for` fields for `PrometheusRule` CRD. #5651
```

> * [FEATURE] Allow to add custom annotations to resources managed by operator #5626

Reviewer: (nit)
```suggestion
* [FEATURE] Allow to add custom annotations to all resources managed by the operator. #5626
```

* [FEATURE] Add `BasicAuth` and `Authorization` support to `ScrapeConfig` CRD #5642

> * [ENHANCEMENT] Add label selector to filter nodes monitored by the Operator #5641

Reviewer: ```suggestion
* [ENHANCEMENT] Add `-kubelet-selector` CLI argument to filter the Kubernetes nodes by labels. #5641
```

* [BUGFIX] Fix bug with logs that don't end with new line(`\n`) characters. #5566
* [BUGFIX] Prometheus and Alertmanager now pickup certificates after renewal #5535

> * [BUGFIX] Config-Reloader now gracefully handles SIGTERM signals #5617

Reviewer: ```suggestion
* [BUGFIX] Fix config-reloader not handling SIGTERM signal. #5617
```
